### PR TITLE
Don't send a canceled context to Unlock

### PR DESCRIPTION
**Pull request by klauspost**
## Description

AFAICT we send a canceled context to unlock (and thereby releaseAll). This will cause network calls to fail.

Instead use background and add 30s timeout.

@vadmeste You know more about this than I. It seems strange this has been undetected, so I tend to not believe it is the issue. I would expect it to be a bigger issue if this is the case, so I am probably wrong.

PTAL.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by vadmeste — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by vadmeste** [comment]
@klauspost I think this did not cause a big issue before because the locking subsystem has a refresh mechanism. That mechanism will clear a lock when its owner does not claim its ownership to that lock anymore (just lock uuid not found in its memory). Since the node will call Lock() then Unlock() it, the unlock will be always successful in that node since it is just a memory operation (deleting the lock uuid from the memory)

So to summarize, the locking refresh mechanism will clear this automatically after 30 seconds.

**Comment by harshavardhana** [comment]
> @klauspost I think this did not cause a big issue before because the locking subsystem has a refresh mechanism. That mechanism will clear a lock when its owner does not claim its ownership to that lock anymore (just lock uuid not found in its memory). Since the node will call Lock() then Unlock() it, the unlock will be always successful in that node since it is just a memory operation (deleting the lock uuid from the memory)

I'm even there, but it's holding it longer than it should. Other propagation delays may be adding to faster-failed refresh() unlocks().

I haven't been fortunate enough to reproduce it at the moment. It looks like you need some competing lockers as well as repeated retries, etc.
